diff --git a/edge-service-api/controller/promocionesController.py b/edge-service-api/controller/promocionesController.py
index 82f3c00..d22cc95 100755
--- a/edge-service-api/controller/promocionesController.py
+++ b/edge-service-api/controller/promocionesController.py
@@ -1,4 +1,4 @@
-import datetime
+from datetime import datetime
 from models.promocionEstacion import PromocionEstacion, PromocionEstacionSchema
 from models.promociones import Promocion, PromocionSchema
 from utils.db import db
@@ -10,38 +10,46 @@ def get_all_promociones():
 
 
 def get_promo_id(id_promo):
-    p = Promocion.query.filter(Promocion.id_promo == id_promo).one_or_none()
+    p = Promocion.query.get(id_promo)
     return PromocionSchema().dump(p)
 
 
-def post_promociones(descuento, fecha_inicio, fecha_fin, estado, descripcion):
+def post_promociones(descuento, fecha_inicio_post, fecha_fin_post, estado, descripcion):
     try:
         # Pasamos a datetime las fechas
-        fecha_inicio = datetime.strptime(fecha_inicio, '%Y-%m-%dT%H:%M:%S')
-        fecha_fin = datetime.strptime(fecha_fin, '%Y-%m-%dT%H:%M:%S')
+        fecha_inicio = datetime.strptime(fecha_inicio_post, '%Y-%m-%dT%H:%M:%S')
+        fecha_fin = datetime.strptime(fecha_fin_post, '%Y-%m-%dT%H:%M:%S')
         p = Promocion(descuento, fecha_inicio, fecha_fin, estado, descripcion)
         db.session.add(p)
         db.session.commit()
         return PromocionSchema().dump(p)
-    except:  # noqa: E722
+    except (ValueError):  # noqa: E722
         return None
 
 
 def modify_promociones(id_promo, descuento=None, fecha_inicio=None, fecha_fin=None, estado=False, descripcion=None):
-    p = Promocion.query.filter(Promocion.id_promo == id_promo).one_or_none()
-    if p:
-        if descuento:
-            p.descuento = descuento
-        if fecha_inicio:
-            p.fecha_inicio = fecha_inicio
-        if fecha_fin:
-            p.fecha_fin = fecha_fin
-        if estado:
-            p.estado = estado
-        if descripcion:
-            p.descripcion = descripcion
-        db.session.commit()
-        return PromocionSchema().dump(p)
+    try:
+        p = get_promo_id(id_promo)
+        p = Promocion(p["descuento"], p["fecha_inicio"], p["fecha_fin"], p["estado"], p["descripcion"])
+        if p:
+            if descuento:
+                p.descuento = descuento
+            if fecha_inicio:
+                p.fecha_inicio = datetime.strptime(fecha_inicio, '%Y-%m-%dT%H:%M:%S')
+            if fecha_fin:
+                p.fecha_fin = datetime.strptime(fecha_fin, '%Y-%m-%dT%H:%M:%S')
+            if estado == 'true':
+                p.estado = True
+            else:
+                p.estado = False
+            if descripcion:
+                p.descripcion = descripcion
+            db.session.commit()
+            return PromocionSchema().dump(p)
+    except (KeyError):
+        return post_promociones(descuento, fecha_inicio, fecha_fin, estado, descripcion)
+    except (ValueError):
+        return None
     return None
 
 
diff --git a/edge-service-api/models/promociones.py b/edge-service-api/models/promociones.py
index 0664e01..35267fb 100755
--- a/edge-service-api/models/promociones.py
+++ b/edge-service-api/models/promociones.py
@@ -6,17 +6,13 @@ class Promocion(db.Model):
 
     id_promo = db.Column(db.Integer, nullable=False, primary_key=True, autoincrement=True)
     descuento = db.Column(db.Integer, nullable=False)
-    cantidad_cupones = db.Column(db.Integer, nullable=False)
-    cantidad_usados = db.Column(db.Integer, nullable=False)
     fecha_inicio = db.Column(db.DateTime, nullable=False)
     fecha_fin = db.Column(db.DateTime, nullable=False)
     estado = db.Column(db.String(30), nullable=False)
     descripcion = db.Column(db.String(300), nullable=False)
 
-    def __init__(self, descuento, cantidad_cupones, cantidad_usados, fecha_inicio, fecha_fin, estado, descripcion):
+    def __init__(self, descuento, fecha_inicio, fecha_fin, estado, descripcion):
         self.descuento = descuento
-        self.cantidad_cupones = cantidad_cupones
-        self.cantidad_usados = cantidad_usados
         self.fecha_inicio = fecha_inicio
         self.fecha_fin = fecha_fin
         self.estado = estado
diff --git a/edge-service-api/routes/promociones.py b/edge-service-api/routes/promociones.py
index 6cbf9dc..638336c 100755
--- a/edge-service-api/routes/promociones.py
+++ b/edge-service-api/routes/promociones.py
@@ -26,8 +26,8 @@ def post_promocion():
 
 
 @promociones.route('/promociones/<id_promocion>', methods=['GET'])
-def get_promo_id(id_promo):
-    respuesta = control.get_promo_id(id_promo)
+def get_promo_id(id_promocion):
+    respuesta = control.get_promo_id(id_promocion)
     if respuesta:
         return jsonify(respuesta), 200
     else:
@@ -35,7 +35,7 @@ def get_promo_id(id_promo):
 
 
 @promociones.route('/promociones/<id_promocion>', methods=["PUT"])
-def modify_promocion(id_promo):
+def modify_promocion(id_promocion):
     descuento = None
     fecha_inicio = None
     fecha_fin = None
@@ -53,7 +53,7 @@ def modify_promocion(id_promo):
     if "descripcion" in request.form.to_dict():
         descripcion = request.form.to_dict()["descripcion"]
 
-    respuesta = control.modify_promociones(id_promo, descuento, fecha_inicio, fecha_fin, estado, descripcion)
+    respuesta = control.modify_promociones(id_promocion, descuento, fecha_inicio, fecha_fin, estado, descripcion)
 
     if respuesta:
         return jsonify(respuesta), 200
diff --git a/edge-service-api/test.db b/edge-service-api/test.db
index 6da5a08..8f86abc 100644
Binary files a/edge-service-api/test.db and b/edge-service-api/test.db differ
